### ISSUE TITLE
Wrong spectral wavenumbers from a hyperspectra. 

### ISSUE THREAD

**Issue by dbedolla** [Quasars/orange-spectroscopy#457]
Hello, I am trying to identify interesting areas by using a PCA on a hyperspectra, but when I try to look at the loadings, wavenumbers do not correspond. Is this a bug? Or am I doing something wrong? My file's spectral range is 4000-850 cm-1, not from 0-2400 (as it is showing). Is there a way to tell which columns it should read for the spectra?

![image](https://user-images.githubusercontent.com/58698691/85745585-19dfb700-b706-11ea-8920-285974cdbce0.png)
![image](https://user-images.githubusercontent.com/58698691/85746016-6fb45f00-b706-11ea-9f30-bceee4bc3b68.png)

Thanks in advance!

**Comment by stuart-cls** [comment]
Hi, I can't reproduce this with just the PCA widget:

![image](https://user-images.githubusercontent.com/19228847/85759626-5c859d00-b6ce-11ea-8510-9ed4623692e8.png)


I wonder if you have some transformation in your **Feature Constructor** or **Integrate Spectra** widgets. Can you check the outputs of those?

**Comment by dbedolla** [comment]
Hi Stuart,

To obtain the loadings, I usually do as you're showing. My problem is when I added the feature constructor and a column is added (X1). When I see the table from the PCA, it shows the correct wavenumbers (3937-850) and the extra column (X1), but when I plot the spectra from that table it plots with different wavenumbers (0-2403), although I believe the y-data is correct. In the attached image, I am showing the table and the spectra. Thanks for taking interest! ;-)

![image](https://user-images.githubusercontent.com/58698691/85843778-fbc79480-b7a1-11ea-9849-8fd05a9ea5fd.png)

**Comment by AlexHenderson** [comment]
It looks as though you're plotting the loading against the channel number (row number), rathr than the wavenumber.

**Comment by dbedolla** [comment]
Hi Alex, 
I think you are right, instead of plotting the wavenumber it is plotting the column index/number. Do you have any idea how to overcome this? Or is this a bug? Thanks!

**Comment by markotoplak** [comment]
@dbedolla, I think that additional column X1 confuses the Spectra widget, because now not all columns are numbers. Moving X1 into Meta Attributes with Select Columns will probably help.

**Comment by dbedolla** [comment]
Yeah, that helped! Thanks a lot!  For this case, I think it would be better if the feature constructor would assign the feature automatically as a Meta Attribute.

**Comment by stuart-cls** [comment]
@dbedolla Glad you figured it out!

@markotoplak is it possible to pop a warning when `orangecontrib.spectroscopy.data.getx()` falls back to indices? I guess we use it so many places they would all have to handle it so it might be too big a job.
